modules/context/repo.go

```diff
@@ -821,3 +835,56 @@ func UnitTypes() macaron.Handler {
 		ctx.Data["UnitTypeProjects"] = models.UnitTypeProjects
 	}
 }
+
+// IssueTemplatesFromDefaultBranch checks for issue templates in the repo's default branch
+func (ctx *Context) IssueTemplatesFromDefaultBranch() []api.IssueTemplate {
+	var issueTemplates []api.IssueTemplate
+	if ctx.Repo.Commit == nil {
+		var err error
+		ctx.Repo.Commit, err = ctx.Repo.GitRepo.GetBranchCommit(ctx.Repo.Repository.DefaultBranch)
+		if err != nil {
+			return issueTemplates
+		}
+	}
+
+	for _, dirName := range IssueTemplateDirCandidates {
+		tree, err := ctx.Repo.Commit.SubTree(dirName)
+		if err != nil {
+			continue
+		}
+		entries, err := tree.ListEntries()
+		if err != nil {
+			return issueTemplates
+		}
+		for _, entry := range entries {
+			if strings.HasSuffix(entry.Name(), ".md") {
+				r, err := entry.Blob().DataAsync()
```

Comment:
Check size similar to [routers/issue.go](https://github.com/go-gitea/gitea/blob/63f3c42be99cd72e765f7d8abfb8b2500b487855/routers/repo/issue.go#L503-L505)